docs: document Save as for Discover panels in embedded editor (#5712) (#6084)
elastic/docs-content: explore-analyze/discover/save-open-search.md

By default, a Discover session stores the query text, filters, and current view of **Discover**, including the columns and sort order in the document table, and the {{data-source}}.

1. Once you’ve created a view worth saving, select **Save** in the application menu.
2. Enter a **Title** for the session, and optionally a **Description** and [**Tags**](../find-and-organize/tags.md).
3. If the session is time-based, turn on **Store time with Discover session** to save the current time filter and refresh interval with it.

> 4. {applies_to}`stack: ga 9.4` {applies_to}`serverless: ga` Under **Add to dashboard**, select **Existing** to add the session as a panel on a dashboard you choose, **New** to add it to a brand-new dashboard, or **None** to save the session to the library only.

Reviewer: Not clear from where **Add to dashboard** is on the UI. I found something similar for the table:

<img width="655" height="345" alt="Image" src="https://github.com/user-attachments/assets/73e1967d-68f5-439f-93bf-3fc79dfc9740" />
I landed here by clicking the "Save table to dashboard" button 

<img width="35" height="31" alt="Image" src="https://github.com/user-attachments/assets/fdb76ea5-5a9f-4335-a63c-0d039fe5dc1e" />

Author: That's part of the main Save modal (step 1) for new discover sessions. I nested the steps to make this more obvious

Author: What you tested is a different workflow also new for 9.4 but documented in a different PR

> 4. {applies_to}`stack: ga 9.4` {applies_to}`serverless: ga` Under **Add to dashboard**, select **Existing** to add the session as a panel on a dashboard you choose, **New** to add it to a brand-new dashboard, or **None** to save the session to the library only.

Author: ```suggestion
1. Once you’ve created a view worth saving, select **Save** in the application menu. A modal with several options opens:
    1. Enter a **Title** for the session, and optionally a **Description** and [**Tags**](../find-and-organize/tags.md).
    2. If the session is time-based, turn on **Store time with Discover session** to save the current time filter and refresh interval with it.
    3. {applies_to}`stack: ga 9.4` {applies_to}`serverless: ga` Under **Add to dashboard**, select **Existing** to add the session as a panel on a dashboard you choose, **New** to add it to a brand-new dashboard, or **None** to save the session to the library only.
```

5. Select **Save**.

> 6. To reload your search results in **Discover**, select **Open** in the application menu, and select the saved Discover session.

Author: ```suggestion
2. Select **Save**.
3. To reload your search results in **Discover**, select **Open** in the application menu, and select the saved Discover session.
```

   :::

3. Select **Apply**.

### Save edits as a new Discover session [discover-session-save-as-from-dashboard]
```{applies_to}
stack: ga 9.4
serverless: ga
```

When editing a Discover session panel from a dashboard, you can save your changes as a new Discover session instead of updating the panel. This works whether the panel was added from the library or created directly on the dashboard.

> 1. From the dashboard, open the panel menu and select **Edit**.

Reviewer: I tried to reproduce these steps in the UI, but it's not obvious. I might not be an expert with Discover but I feel like something is missing.

Author: I had no problem following these exact steps, can you tell me more about what was not obvious?

Author: I slightly rephrased step 1 there if that was the issue

> 1. From the dashboard, open the panel menu and select **Edit**.

Author: ```suggestion
1. From the dashboard, hover over the panel and select {icon}`pencil` **Edit Discover session configuration**.
```

2. Make your changes in **Discover**.
3. In the application menu, open the menu next to **Save and return** and select **Save as**.

> 4. In the **Save Discover session** window, enter a **Title** for the new session, and optionally a **Description** and [**Tags**](../find-and-organize/tags.md).

Reviewer 2: ```suggestion
4. In the **Save Discover session** modal, enter a **Title** for the new session, and optionally a **Description** and [**Tags**](../find-and-organize/tags.md).
```
Nit: would "modal" be more standard here?

Author: Yes totally!

> 5. Under **Add to dashboard**, choose where to display the new session:

Author: ```suggestion
5. In **Add to dashboard**, choose where to display the new session:
```

   - **Existing**: Add the new session to a dashboard you select.

>      - If you select the dashboard the panel came from, the original panel is updated in place to reference the new session, in the same position. If the panel was previously linked to a saved Discover session in the library, that library session is unchanged. If the panel held its state directly on the dashboard (not linked to the library), that inline state is replaced by the link to the new session.

Author: ```suggestion
     - If you select the dashboard the panel came from, the original panel is updated in place to reference the new session, in the same position. If the replaced panel was linked to the library, you can still find it unchanged in the library. If the panel wasn't linked to the library, it is lost and replaced by the newly saved session.
```

>      - If you select a different dashboard, the original panel is unchanged, and the new session is added as a separate panel on the dashboard you selected.

Reviewer 2: While technically accurate, I wonder if it's worth publicly documenting this bug 😅 We intend to fix it soon anyway, but I guess we could keep this until then if preferred.

Author: Yes I think it's safer to document it for now, and hopefully we can remove it at 9.4.1

   - **New**: Save the session and add it as a panel on a new dashboard. The original panel is unchanged.
   - **None**: Save the session to the library only, without adding it to a dashboard. The original panel is unchanged.

6. Select **Save and add to library** (when **None** is selected) or **Save and go to Dashboard** (when **Existing** or **New** is selected).

> The embedded editor closes. You land on the new session in **Discover** when you selected **None**, or on the chosen dashboard when you selected **Existing** or **New**.

Author: ```suggestion
Upon saving, you navigate to the new session in **Discover** when you selected **None**, or to the corresponding dashboard when you selected **Existing** or **New**.
```
